### PR TITLE
Added job profiles related to infrastructure

### PR DESCRIPTION
**Pull request by meysholdt**
This PR adds job profiles for 

* Senior Software Engineer (Infrastructure): Terraform, Helm, Cloud-Integration
* Senior Software Engineer (Observability): from creating metrics, logs, traces, etc. to storing and exploring them.
* Kubernetes Engineer: self-managed Kubernetes, Kubernetes integration.
* Site Reliability Engineer (SRE): master of incidents

Status:
* I'm quite happy with the role definitions.
* The tag lines can be shorter and more on-point
* The "You will be a great fit if ..." are not enough on-point.

### CODE REVIEW

**Review by JohannesLandgraf — COMMENTED** [review]
Some small comments - but otherwise looks good 👍

File: src/pages/careers.tsx [hunk 1]
```diff
@@ -267,43 +267,191 @@ const jobs: ExpandableJobProps[] = [
...
+                    'Monthly physical health stipend for your local gym, home-gym equipment or Freeletics',
+                    'Monthly mental health stipend to support your 🧘‍♀️',
+                    'Blinkist, The Atlantic account and other educational budget',
+                    'Premium work-from-home equipment (Macbook Pro, standing desk, monitor, webcam, mic)',
+                ]
+            },
+        ],
+        textAfterTheLists: '',
+        rendered: false,
+        date: '2021-19-01T19:00:00.000Z'
```
Comment (jankeromnes) [inline src/pages/careers.tsx:318]: Nit: FYI there still isn't a 19th month in the year 2021

Comment (jankeromnes) [reply src/pages/careers.tsx:318]: Fixed: https://github.com/gitpod-io/website/commit/66699f922153506d97056960d45a48f0036c501e

Comment (JohannesLandgraf) [reply src/pages/careers.tsx:318]: Haha fair point

**Pull request merged**